contracts/base_contracts.py

```diff
@@ -0,0 +1,325 @@
+        if not raw_results:
+            return 0
+        result = int(raw_results)
+
+        return result
+    
+    def get_last_global_update_week(self, proxy: ProxyNetworkProvider) -> int:
+        data_fetcher = BaseBoostedContractDataFetcher(Address(self.address), proxy.url)
+        raw_results = data_fetcher.get_data('getLastGlobalUpdateWeek')
+        if not raw_results:
+            return 0
+        result = int(raw_results)
+
+        return result
+    
+    def get_user_energy_for_week(self, user_address: str, proxy: ProxyNetworkProvider, week: int) -> Dict[str, Any]:
+        data_fetcher = BaseBoostedContractDataFetcher(Address(self.address), proxy.url)
+        raw_results = data_fetcher.get_data('getUserEnergyForWeek', [Address(user_address).serialize(), week])
+        if not raw_results:
+            return {}
+        user_energy_for_week = decode_merged_attributes(raw_results, decoding_structures.ENERGY_ENTRY)
+
+        return user_energy_for_week
+    
+    def get_last_active_week_for_user(self, user_address: str, proxy: ProxyNetworkProvider) -> int:
+        data_fetcher = BaseBoostedContractDataFetcher(Address(self.address), proxy.url)
+        raw_results = data_fetcher.get_data('getLastActiveWeekForUser', [Address(user_address).serialize()])
+        if not raw_results:
+            return 0
+        week = int(raw_results)
+
+        return week
+    
+    def get_current_claim_progress_for_user(self, user_address: str, proxy: ProxyNetworkProvider) -> Dict[str, Any]:
+        data_fetcher = BaseBoostedContractDataFetcher(Address(self.address), proxy.url)
+        raw_results = data_fetcher.get_data('getCurrentClaimProgress', [Address(user_address).serialize()])
+        if not raw_results:
+            return {}
+        response = decode_merged_attributes(raw_results, decoding_structures.USER_CLAIM_PROGRESS)
+
+        return response
+    
+    def get_farm_supply_for_week(self, proxy: ProxyNetworkProvider, week: int) -> int:
+        data_fetcher = BaseBoostedContractDataFetcher(Address(self.address), proxy.url)
+        raw_results = data_fetcher.get_data('getFarmSupplyForWeek', [week])
+        if not raw_results:
+            return 0
+        return int(raw_results)
+    
+    def get_total_locked_tokens_for_week(self, proxy: ProxyNetworkProvider, week: int) -> int:
+        data_fetcher = BaseBoostedContractDataFetcher(Address(self.address), proxy.url)
+        raw_results = data_fetcher.get_data('getTotalLockedTokensForWeek', [week])
+        if not raw_results:
+            return 0
+        return int(raw_results)
+    
+    def get_total_energy_for_week(self, proxy: ProxyNetworkProvider, week: int) -> int:
+        data_fetcher = BaseBoostedContractDataFetcher(Address(self.address), proxy.url)
+        raw_results = data_fetcher.get_data('getTotalEnergyForWeek', [week])
+        if not raw_results:
+            return 0
+        return int(raw_results)
```

Comment:
does it work for large BigUint number?

Reply:
Yes, int in python 3 can store arbitrarily large numbers.